protocols/kad/src/behaviour.rs

```diff
@@ -2031,6 +1990,34 @@ where
             self.connected_peers.remove(&peer_id);
         }
     }
+
+    /// Preloads a new [`Handler`] with requests that are waiting to be sent to the newly connected peer.
+    fn preload_new_handler(
+        &mut self,
+        handler: &mut Handler,
+        connection_id: ConnectionId,
+        peer: PeerId,
+    ) {
+        // Queue events for sending pending RPCs to the connected peer.
+        // There can be only one pending RPC for a particular peer and query per definition.
+        for (peer_id, event) in self.queries.iter_mut().filter_map(|q| {
+            q.inner
+                .pending_rpcs
+                .iter()
+                .position(|(p, _)| p == &peer)
+                .map(|p| q.inner.pending_rpcs.remove(p))
+        }) {
+            self.queued_events.push_back(ToSwarm::NotifyHandler {
+                peer_id,
+                event: event.clone(),
+                handler: NotifyHandler::Any,
+            });
```

Comment:
There is no need to push the events to `queued_events` if we are handing them to the handler!